packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -125,14 +129,16 @@ class Scrollable2State extends State<Scrollable2> with TickerProviderStateMixin
     if (config.physics != null)
       physics = config.physics.applyTo(physics);
     final ScrollPosition oldPosition = position;
-    _position = physics.createScrollPosition(physics, this, oldPosition);
-    assert(position != null);
     if (oldPosition != null) {
+      config.controller?.detach(oldPosition);
       // It's important that we not do this until after the viewport has had a
```

Comment:
while you're here can you clarify my comment here by changing "do this" to "dispose of the old position"? I just got confused reading my own comment...